fairlearn/datasets/_fetch_acsincome.py

```diff
@@ -0,0 +1,143 @@
+from ._constants import _DOWNLOAD_DIRECTORY_NAME
+
+
+def fetch_acsincome(*, cache=True, data_home=None,
+                    as_frame=False, return_X_y=False,
+                    states=None,
+                    ):
+    """Load the ACS Income dataset.
+
+    Download it if necessary.
+
+    ==============   ==============
+    Samples total           1664500
+    Dimensionality               10
+    Features                   real
+    Target                     real
+    ==============   ==============
+
+    Source: Paper: Ding et al. (2021) [1]_
+            and corresponding repository https://github.com/zykls/folktables/
+
+    .. versionadded:: 0.7.1
+
+    Parameters
+    ----------
+    cache : bool, default=True
+        Whether to cache downloaded datasets using joblib.
+
+    data_home : str, default=None
+        Specify another download and cache folder for the datasets.
+        By default, all fairlearn data is stored in '~/.fairlearn-data'
+        subfolders.
+
+    as_frame : bool, default=False
+        If True, the data is a pandas DataFrame including columns with
+        appropriate dtypes (numeric, string or categorical). The target is
+        a pandas DataFrame or Series depending on the number of target_columns.
+        The Bunch will contain a ``frame`` attribute with the target and the
+        data. If ``return_X_y`` is True, then ``(data, target)`` will be pandas
+        DataFrames or Series as describe above.
+
+    return_X_y : bool, default=False
+        If True, returns ``(data.data, data.target)`` instead of a Bunch
+        object.
+
+    states: list, default=None
+        List containing two letter (capitalized) state abbreviations.
+        If None, data from all 50 US states, including Puerto Rico, will be returned.
+        The state abbreviations and codes can be found on page 1 of the data dictionary at ACS PUMS
+        https://www2.census.gov/programs-surveys/acs/tech_docs/pums/data_dict/PUMS_Data_Dictionary_2018.pdf
+        and are reproduced in the `_STATE_CODES` dictionary below.
+
+    Returns
+    -------
+    dataset : :obj:`~sklearn.utils.Bunch`
+        Dictionary-like object, with the following attributes.
+
+        data : ndarray, shape (1664500, 10)
+            Each row corresponding to the 10 feature values in order.
+            If ``as_frame`` is True, ``data`` is a pandas object.
+        target : numpy array of shape (1664500,)
+            Integer denoting each person's annual income.
+            A threshold can be applied as a postprocessing step to frame this
+            as a binary classification problem.
+            If ``as_frame`` is True, ``target`` is a pandas object.
+        feature_names : list of length 10
+            Array of ordered feature names used in the dataset.
+        DESCR : string
+            Description of the ACSIncome dataset.
+
+    (data, target) : tuple of (numpy.ndarray, numpy.ndarray) or (pandas.DataFrame, pandas.Series)
```

Comment:
This might be a copy-paste error, I think this should only return a tuple of (numpy.ndarray, numpy.ndarray) since `as_frame` is False?

Reply:
Good catch. I'll fix this in the other files as well